azext_iot/operations/digitaltwin.py

```diff
@@ -0,0 +1,275 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Unpublished works.
+# --------------------------------------------------------------------------------------------
+
+from os.path import exists
+from knack.util import CLIError
+from azure.cli.core.util import read_file_content
+from azext_iot._constants import PNP_ENDPOINT
+from azext_iot._factory import _bind_sdk
+from azext_iot.common.shared import SdkType, ModelSourceType
+from azext_iot.common._azure import get_iot_hub_connection_string
+from azext_iot.common.utility import (shell_safe_json_parse,
+                                      unpack_msrest_error)
+from azext_iot.operations.pnp import (iot_pnp_interface_show,
+                                      iot_pnp_interface_list,
+                                      _validate_repository)
+from azext_iot.operations.hub import _iot_hub_monitor_events
+
+
+INTERFACE_KEY_NAME = 'urn_azureiot_ModelDiscovery_DigitalTwin'
+INTERFACE_COMMAND = 'Command'
+INTERFACE_PROPERTY = 'Property'
+INTERFACE_TELEMETRY = 'Telemetry'
+INTERFACE_MODELDEFINITION = 'urn_azureiot_ModelDiscovery_ModelDefinition'
+INTERFACE_COMMANDNAME = 'getModelDefinition'
+
+
+def iot_digitaltwin_interface_list(cmd, device_id, hub_name=None, resource_group_name=None, login=None):
+    device_default_interface = _iot_digitaltwin_interface_show(cmd, device_id, INTERFACE_KEY_NAME,
+                                                               hub_name, resource_group_name, login)
+    result = _get_device_default_interface_dict(device_default_interface)
+    return {'interfaces': result}
+
+
+def iot_digitaltwin_command_list(cmd, device_id, source_model, interface=None, schema=False,
+                                 repo_endpoint=PNP_ENDPOINT, repo_id=None, repo_login=None,
+                                 hub_name=None, resource_group_name=None, login=None):
+    result = []
+    target_interfaces = []
```

Comment:
Consider adding `source_model = source_model.lower()` here, so you only `lower()` once on it.

Reply:
Done